lib/secure_headers.rb

```diff
@@ -190,17 +270,19 @@ def header_classes_for(request)
         ALL_HEADER_CLASSES
       else
         HTTP_HEADER_CLASSES
-      end
+      end.map do |klass|
```

Comment:
This is going to take another non-trivial, but overdue refactoring of the way CSP vs CSPRO are handled. The result should be much cleaner. Passing around symbols and class objects with all sorts of special casing have trickled down.